salt/cloud/clouds/libvirt.py

```diff
@@ -441,37 +594,203 @@ def create(vm_):
+                        )
+                        devices_xml.append(
+                            ElementTree.Element("disk", type="file", device="disk")
+                        )
+                        disk_xml = devices_xml.findall("./disk")[-1]
+                        disk_xml.append(
+                            ElementTree.Element("driver", name="qemu", type="raw")
+                        )
+                        disk_xml.append(ElementTree.Element("source", file=device))
+                        disk_xml.append(
+                            ElementTree.Element("target", dev=disk, bus=bus)
+                        )
+                        if shareable:
+                            disk_xml.append(ElementTree.Element("shareable"))
+
+                    else:
+                        # format: should be 'raw' or 'qcow2' (default)
+                        if "format" in devices["disk"][disk]:
+                            format = devices["disk"][disk]["format"]
+                        else:
+                            format = "qcow2"
+
+                        # pool: name of the libvirt pool that will contain the new disk (default is "default")
+                        if "pool" in devices["disk"][disk]:
+                            pool_name = devices["disk"][disk]["pool"]
+                        else:
+                            pool_name = "default"
+
+                        # size: should be a size in GB (default: 1 GB)
+                        if "size" in devices["disk"][disk]:
+                            size = devices["disk"][disk]["size"]
+                        else:
+                            size = 1
+
+                        # thin_provision: wether the disk will be thin provsionned or fully allocated (default)
+                        if "thin_provision" in devices["disk"][disk]:
+                            thin_provision = devices["disk"][disk]["thin_provision"]
+                        else:
+                            thin_provision = False
+                        if thin_provision:
+                            volumeFlags = 0
+                        else:
+                            volumeFlags = (
+                                libvirt.VIR_STORAGE_VOL_CREATE_PREALLOC_METADATA
+                            )
+
+                        pool = None
+                        for p in virt_pools:
+                            if p.name() == devices["disk"][disk]["pool"]:
+                                pool = p
+
+                        if pool:
+                            pool_xml = ElementTree.fromstring(pool.XMLDesc())
+                            pool_target = pool_xml.find("./target/path").text
+
+                            vol_name = name + "-" + disk + "." + format
+                            vol_path = pool_target + "/" + vol_name
+
+                            try:
+                                vol = pool.storageVolLookupByName(vol_name)
```

Comment:
You surely want to refresh the pool here. Maybe even ensure it is started...

Reply:
OK, I've added the pool.refresh() and pool.create()

Reply:
thanks